flang/lib/Lower/NumericRuntime.cpp

```diff
@@ -0,0 +1,102 @@
+//===-- NumericRuntime.cpp -- runtime for numeric intrinsics -------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "../../runtime/numeric.h"
+#include "RTBuilder.h"
+#include "flang/Lower/Bridge.h"
+#include "flang/Lower/CharacterExpr.h"
+#include "flang/Lower/FIRBuilder.h"
+#include "flang/Lower/Support/BoxValue.h"
+#include "flang/Lower/Todo.h"
+#include "mlir/Dialect/StandardOps/IR/Ops.h"
+#include "flang/Lower/NumericRuntime.h"
+
+using namespace Fortran::runtime;
+
+/// Helper function to recover the KIND from the REAL FIR type.
+static int discoverRealKind(mlir::Type fltTy) {
```

Comment:
@schweitzpgi , isn't there a way to do that with the KindMapping  from `builder.getKindMap()` ?

Reply:
We normally convert KIND values to target types (the other direction), so, no. (It may not always be possible as the KIND map could be surjective.)

Reply:
I don't think this function is needed.